samples/JobsAgent/JobsAgent.hpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/**
+ * @file JobsAgent.hpp
+ * @brief
+ *
+ */
+
+
+#pragma once
+
+#include "mqtt/Client.hpp"
+#include "NetworkConnection.hpp"
+#include "jobs/Jobs.hpp"
+#include <curl/curl.h>
+#ifdef UNIT_TESTS
+#include "JobsMock.hpp"
+#endif
+
+#define LOG_TAG_JOBS_AGENT "[Sample - JobsAgent]"
+
+namespace awsiotsdk {
+    namespace samples {
+        class JobsAgent {
+        protected:
+            static const util::String INSTALLED_PACKAGES_FILENAME;
```

Comment:
This is typically bad practice. static scope allocations are non-deterministic in order and if a custom memory allocator is used, this will crash it.